tensorflow-core/tensorflow-core-api/src/main/java/org/tensorflow/op/NameScope.java

```diff
@@ -50,6 +53,45 @@ NameScope withName(String name) {
     return new NameScope(opPrefix, name, ids);
   }
 
+  private static final Pattern NAME_PATTERN = Pattern.compile("(.+)_(\\d+)", Pattern.DOTALL);
+
+  /**
+   * "Import" used names from a graph.  Useful when adding to a loaded graph.
+   */
+  NameScope withUsedFrom(ExecutionEnvironment env) {
```

Comment:
Since calling `withUsedFrom` seems a bit mandatory everytime you want to instantiate a new `NameScope`, why not passing the `env` to its constructor and its `withSubScope` methods so we never forget it?